zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -165,8 +165,11 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
     implicit def trace: Trace = InteropTracer.newTrace(cb)
 
     effect.SyncIO {
-      rts.unsafeRunAsyncWith(fa.exit) { exit =>
-        cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
+      Unsafe.unsafeCompat { implicit u =>
+        val fiber = rts.unsafe.fork(fa.exit)
+        fiber.unsafe.addObserver { exit =>
+          cb(exit.getOrThrowFiberFailure().toEither).unsafeRunAsync(_ => ())
+        }
```

Comment:
i get the same test results with the (slightly simpler)
```scala
      Unsafe.unsafe { implicit u =>
        rts.unsafe.fork(fa).unsafe.addObserver { exit =>
          cb(exit.toEither).unsafeRunAsync(_ => ())
        }
```
but again, not an expert